frontend/src/components/HighlightedText.tsx

```
@@ -0,0 +1,264 @@
+    // Check window exists for environments like SSR
+    return typeof window !== 'undefined' && window.matchMedia('(hover: none)').matches;
+  }, []);
+
+  const segments = useMemo(() => {
+    if (!text || !selections.length) {
+      return [{ start: 0, end: text.length, text, overlapCount: 0, overlappingQuotes: [] }];
+    }
+
+    let boundaries: number[] = [0, text.length];
+    selections.forEach(quote => {
+      if (quote && quote.selectionRange) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)